src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -53,17 +49,26 @@ private bool TrySetBasicAuthToken(HttpRequestMessage request)
 
                 foreach (AuthenticationHeaderValue h in authenticateValues)
                 {
-                    // We only support Basic auth, ignore others
-                    if (h.Scheme == Basic)
+                    // We only support Basic and digest auth, ignore others
```

Comment:
By something odd, I mean something which can cause problems like endless loops. If you get told your nonce is stale, you only get one chance at it to prevent malicious servers causing problems.